src/confcom/azext_confcom/custom.py

```diff
@@ -245,24 +245,28 @@ def acifragmentgen_confcom(
         elif image_name:
             import_statements = oras_proxy.generate_imports_from_image_name(image_name, minimum_svn=minimum_svn)
 
+        fragments_file_contents = {}
+        fragments_list = []
         if fragments_json:
+            logger.info("Creating/appending import statement JSON file")
             if os.path.isfile(fragments_json):
-                logger.info("Appending import statement to JSON file")
                 fragments_file_contents = os_util.load_json_from_file(fragments_json)
-                fragments_list = fragments_file_contents.get(POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS, [])
-            else:
-                logger.info("Creating import statement JSON file")
-                fragments_file_contents = {}
-                fragments_list = []
-            # convert to list if it's just a dict
-            if not isinstance(fragments_list, list):
-                fragments_list = [fragments_list]
-            fragments_list += import_statements
-
-            fragments_file_contents[POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS] = fragments_list
+                try:
+                    fragments_list = fragments_file_contents.get(POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS, [])
+                except AttributeError:
+                    logger.error("Invalid JSON file format. Please make sure the outermost structure is not an array. An empty import file should look like: %s", REGO_IMPORT_FILE_STRUCTURE)
```

Comment:
I'm kinda lost. JSON supports lists but it looks like we error if it's not a dict? But later on, it looks like we do a conversion to lists if it's just a dict?

Reply:
it does look a little confusing. the value inside `POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS` is a list. the AttributeError happens if fragments_file_contents itself is a list. So the outermost item in the json can't be a list, but inside `POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS` should be a list

Reply:
Ok, so the try-except is equivalent to `isinstance(fragments_file_contents, dict)`? I feel like isinstance would be more natural than try-except.

I guess if arrays are JSON, we should say "unsupported" instead of invalid?

Reply:
instead of
# convert to list if it's just a dict
if not isinstance(fragments_list, list)

do we wanna use isinstance to check if fragments_list is a dict? is there something that's validating that if it's not a list, then it must be a dict